docs/specifications/broker-authentication.md

```diff
@@ -0,0 +1,107 @@
+
+## Open Issues
+
+* reloading secrets is different between kubernetes and openshift. [see updating secrets](#updating-secrets)
+    * might need to reconcile the secret on an interval if not automatic
+* shared secrets are mysterious. [OpenShift docs](https://github.com/openshift/openshift-docs/blob/master/dev_guide/secrets.adoc#creating-secrets) state:
+> Secret data can be shared within a namespace
+Except that the broker and the service-catalog run in different namespaces. How
+do we *share* a secret in this case?
+
+## Relevant information
+
+### Updating secrets
+
+[Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/#creating-your-own-secrets)
+
+> Mounted Secrets are updated automatically
+>
+> When a secret being already consumed in a volume is updated, projected keys are eventually updated as well. Kubelet is checking whether the mounted secret is fresh on every periodic sync. However, it is using its local ttl-based cache for getting the current value of the secret. As a result, the total delay from the moment when the secret is updated to the moment when new keys are projected to the pod can be as long as kubelet sync period + ttl of secrets cache in kubelet
+
+But the latest [OpenShift docs](https://github.com/openshift/openshift-docs/blob/master/dev_guide/secrets.adoc#dev-guide-secrets-using-secrets) states something different entirely:
+
+> When you modify the value of a secret, the value (used by an already running pod) will not dynamically change. To change a secret, you must delete the original pod and create a new pod (perhaps with an identical PodSpec).
+>
+> Updating a secret follows the same workflow as deploying a new container image. You can use the kubectl rolling-update command.
+
+It would be really useful if we could do the automatic updates, then the broker
+could always read the secret and always get the latest.
+
+
+### Broker spec authentication
+The current [OpenServiceBroker API authentication](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#authentication) uses basic auth or nothing. The marketplace in our world is the service catalog.
+> The marketplace MUST authenticate with the service broker using HTTP basic authentication (the Authorization: header) on every request. The broker is responsible for validating the username and password and returning a 401 Unauthorized message if credentials are invalid. It is RECOMMENDED that brokers support secure communication from platform marketplaces over TLS.
+
+
+### Service Catalog Basic Auth
+
+The Service Catalog calls [req.SetBasicAuth(c.username, c.password)](https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/brokerapi/openservicebroker/open_service_broker_client.go#L121) when calling teh broker.
+
+This is done by creating a secret with the username and password in it. Then
```

Comment:
> Do we need to handle revocation of credentials? Any way to issue new credentials? (Sounds like this may be related to secret rotation).

The revocation would happen during the rotation. At least that is how I interpreted it.